### ISSUE TITLE
`ReadonlyDeep` does not work for objects with call signatures

### ISSUE THREAD

**Issue by marekdedic** [sindresorhus/type-fest#337]
Hi,
I have recently stumbled upon an issue with the `ReadonlyDeep<>` type. I have a type which has both members as well as call signatures (in my case the `JQueryStatic` type from `@types/jquery`). When I tried to create `ReadonlyDeep<JQueryStatic>`, what I got back was just `JQueryStatic` - no change in type...

Eventually, I tracked it down to the definition:
https://github.com/sindresorhus/type-fest/blob/f45033c5449652143261cbeaba6f4eff3f73c2a0/source/readonly-deep.d.ts#L37-L45
In this case, because the type has a call signature, `T extends Primitive | ((...arguments: any[]) => unknown)` evaluates to `true` and therefore what you get back is the original type...

## Trying to fix the issue

I tried to fix the issue by changing the first check to be just `T extends Primitive`, but then the type would lose all of its call signatures, as it would return `ReadonlyObjectDeep<>`...

So I tried changing `ReadonlyObjectDeep` from 
https://github.com/sindresorhus/type-fest/blob/f45033c5449652143261cbeaba6f4eff3f73c2a0/source/readonly-deep.d.ts#L62-L64
to
```ts
type ReadonlyObjectDeep<ObjectType extends object> = {
    readonly [KeyType in keyof ObjectType]: ReadonlyDeep<ObjectType[KeyType]>
    (...parameters: Parameters<ObjectType>): ReturnType<ObjectType>
};
```
but that just got me `TS7061: A mapped type may not declare properties or methods.`

Is there any way to fix this? Also, as far as I know TS objects have members, call signatures, construct signatures and index infos - the current `ReadonlyObjectDeep` does only members and index infos, I stumbled here upon call signatures, but construct signatures should probably be solved as well...

**Comment by skarab42** [comment]
Seems to be fixed with #359 ([Playground](https://www.typescriptlang.org/play?ts=4.8.2#code/JYWwDg9gTgLgBDAnmApnA3nASighgEwgDsAbRAERRTDgF84AzKCEOAciVQFoGUBnGGwDcAKBHAiMFFAa4AxmgBiECBhFw4AbQAeALjgCoEgOYBdfYZNwAPnCIBXEACNpojTouJnEEuYNenHzdGFQsYIyJjYKdcKH0HZ1cRWjFONAAhWLgAXjV3PQNwkz9LSJs7RxcocuUIYI9-b19PJujY+MqkjQZQuFrRFJE0vpUcAmIyHOw8QlIKKjAAHlqAPlFhzKgx2cnc7Yn56kXNtdTkDNj9udqpzauyTTYeiDZTUSA))

**Comment by marekdedic** [comment]
Thanks, trying it out!
